### PR TITLE
feat: update `Publish-*` methods for JSON generation

### LINKED ISSUE #37: Add support for JSON output to each `Publish-*` cmdlet
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

For each `Publish-*` cmdlet, add a flag for either JSON or HTML output. 

JSON output is necessary for integration of the `VMware.CloudFoundation.Reporting` module with other tools. 

### Use Case(s)

integration of VMware.CloudFoundation.Reporting into other tools. 

### Potential Configuration

To be determined.

### References

_No response_

### PR DESCRIPTION
**Pull request by bhumitra**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware/powershell-module-for-cloud-foundation-reorting/blob/main/CONTRIBUTING_DCO.md) for making a pull request.

**Summary of Pull Request**

Updated `Publish-*` cmdlets to support JSON generation for:
- `Publish-BackupStatus`
- `Publish-NsxtTransportNodeStatus`
- `Publish-NsxtTier0BgpStatus`
- `Publish-SnapshotStatus`
- `Publish-LocalUserExpiry`
- `Publish-StorageCapacityHealth`

Added 2 additional functions:
- `Publish-NsxtHealthNonSOS`
- `Publish-ComponentConnectivityHealthNonSOS`

Signed-off-by: bhumitra nagar <bnagar@vmware.com>

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style / formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Closes #37

**Test and Documentation Coverage**

- [x] Tests have been completed (for bug fixes / features).
- [ ] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?** 

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

New if/else path has been added to these cmdlets. if called using `-jsonOutput,` then JSON is generated. Else, the functionality is the same as before. I have tested the functionality and it works fine.

### CODE REVIEW

**Comment by tenthirtyam** [comment]
Hi @bhumitra -

I’ve marked the pull request for the `v1.1.0` milestone release planned for February 2023. We will hold the merging of this pull request until after the `v1.0.6` milestone release for January 2023. Reviews will, however, occur soon. 🚀 

**Ryan Johnson**
Senior Staff Solutions Architect | Product Engineering @ VMware, Inc.

File: VMware.CloudFoundation.Reporting.psm1 [hunk 1]
```diff
@@ -2213,6 +2229,11 @@ Function Publish-BackupStatus {
         .EXAMPLE
         Publish-BackupStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -workloadDomain sfo-w01
         This example will publish the backup status for the vCenter Server instances, and NSX Local Manager clusters in Workload Domain sfo-w01.
+
+        .EXAMPLE
+        Publish-BackupStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reports
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2234]:
```suggestion
        Publish-BackupStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reporting
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 2]
```diff
@@ -2213,6 +2229,11 @@ Function Publish-BackupStatus {
...
+        This example will generate a json for the backup status for the vCenter Server instances, and NSX Local Manager clusters in a VMware Cloud Foundation instance
+        and save it under F:\Reports with filename <timestamp>-backup-status.json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2236]:
```suggestion
        and save it under F:\Reporting with filename <timestamp>-backup-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 3]
```diff
@@ -2301,14 +2329,21 @@ Function Publish-NsxtTransportNodeStatus {
         .EXAMPLE
         Publish-NsxtTransportNodeStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -workloadDomain sfo-w01
         This example will publish the BGP status for the NSX transport nodes in a VMware Cloud Foundation instance for a workload domain named sfo-w01.
+
+        .EXAMPLE
+        Publish-NsxtTransportNodeStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2334]:
```suggestion
        Publish-NsxtTransportNodeStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reports
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 4]
```diff
@@ -2301,14 +2329,21 @@ Function Publish-NsxtTransportNodeStatus {
...
+        This example will generate a json for the status of all NSX transport nodes in a VMware Cloud Foundation instance.
+        and save it under F:\Reports with filename <timestamp>-nsxttransportnode-status.json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2336]:
```suggestion
        and save it under F:\Reporting with filename <timestamp>-nsxttransportnode-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 5]
```diff
@@ -2377,14 +2418,23 @@ Function Publish-NsxtTransportNodeTunnelStatus {
         .EXAMPLE
         Publish-NsxtTransportNodeTunnelStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -workloadDomain sfo-w01
         This example will publish the BGP status for the NSX transport node tunnels in a VMware Cloud Foundation instance for a workload domain named sfo-w01.
+
+        .EXAMPLE
+        Publish-NsxtTransportNodeTunnelStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reports
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2423]:
```suggestion
        Publish-NsxtTransportNodeTunnelStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reporting
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 6]
```diff
@@ -2377,14 +2418,23 @@ Function Publish-NsxtTransportNodeTunnelStatus {
...
+        This example will generate a json for the status of all NSX transport node tunnels in a VMware Cloud Foundation instance
+        and save it under F:\Reports with filename <timestamp>-nsxttntunnel-status.json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2425]:
```suggestion
        and saves it under F:\Reporting with filename <timestamp>-nsxttntunnel-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 7]
```diff
@@ -2454,14 +2509,21 @@ Function Publish-NsxtTier0BgpStatus {
         .EXAMPLE
         Publish-NsxtTier0BgpStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -workloadDomain sfo-w01
         This example will publish the BGP status for the NSX Tier-0 gateways in a VMware Cloud Foundation instance for a workload domain names sfo-w01.
+
+        .EXAMPLE
+        Publish-NsxtTier0BgpStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reports
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2514]:
```suggestion
        Publish-NsxtTier0BgpStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reporting
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 8]
```diff
@@ -2454,14 +2509,21 @@ Function Publish-NsxtTier0BgpStatus {
...
+        This example will generate a json for the BGP status for all NSX Tier-0 gateways in a VMware Cloud Foundation instance.
+        and save it under F:\Reports with filename <timestamp>-nsxttier0bgp-status.json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2516]:
```suggestion
        and saves it under F:\Reporting with filename <timestamp>-nsxttier0bgp-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 9]
```diff
@@ -2536,6 +2603,12 @@ Function Publish-SnapshotStatus {
         .EXAMPLE
         Publish-SnapshotStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -workloadDomain sfo-w01
         This example will publish the snapshot status for the SDDC Manager, vCenter Server instance, and NSX Edge nodes managed by SDDC Manager for a workload domain names sfo-w01.
+
+        .EXAMPLE
+        Publish-SnapshotStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reports
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2608]:
```suggestion
        Publish-SnapshotStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reporting
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 10]
```diff
@@ -2536,6 +2603,12 @@ Function Publish-SnapshotStatus {
...
+        This example will generate a json for the snapshot status for the SDDC Manager, vCenter Server instances, and NSX Edge nodes managed by SDDC Manager.
+        and save it under F:\Reports with filename <timestamp>-snapshot-status.json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2610]:
```suggestion
        and saves it under F:\Reporting with filename <timestamp>-snapshot-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 11]
```diff
@@ -2623,6 +2705,12 @@ Function Publish-LocalUserExpiry {
         .EXAMPLE
         Publish-LocalUserExpiry -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -sddcRootPass VMw@re1! -allDomains -failureOnly
         This example checks the expiry for local OS users for all Workload Domains across the VMware Cloud Foundation instance but only reports issues.
+
+        .EXAMPLE
+        Publish-LocalUserExpiry -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -sddcRootPass VMw@re1! -allDomains -outputJson F:\Reports
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2710]:
```suggestion
        Publish-LocalUserExpiry -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -sddcRootPass VMw@re1! -allDomains -outputJson F:\Reporting
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 12]
```diff
@@ -2623,6 +2705,12 @@ Function Publish-LocalUserExpiry {
...
+        This example checks the expiry for local OS users for all Workload Domains across the VMware Cloud Foundation
+        and save it as a json under F:\Reports with filename <timestamp>-localuserexpiry-status.json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2712]:
```suggestion
        and saves it under F:\Reporting with filename <timestamp>-localuserexpiry-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 13]
```diff
@@ -2678,22 +2767,118 @@ Function Publish-LocalUserExpiry {
...
+        .EXAMPLE
+        Publish-NsxtHealthNonSOS -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -workloadDomain sfo-w01
+        This example checks NSX Manager health outside SOS Utility for a single Workload Domain in a VMware Cloud Foundation instance.
+
+        .EXAMPLE
+        Publish-NsxtHealthNonSOS -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -failureOnly
+        This example checks NSX Manager health outside SOS Utility for all Workload Domains across the VMware Cloud Foundation instance but only reports issues.
+
+        .EXAMPLE
+        Publish-NsxtHealthNonSOS -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reports
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2820]:
```suggestion
        Publish-NsxtHealthNonSOS -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reporting
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 14]
```diff
@@ -2678,22 +2767,118 @@ Function Publish-LocalUserExpiry {
...
+        This example checks NSX Manager health outside SOS Utility for all Workload Domains across the VMware Cloud Foundation instance and
+        saves it as a JSON to file under F:\Reports\<timestamp>-nsxtcombinedhealthnonsos-status.json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2822]:
```suggestion
       and saves it as JSON under F:\Reporting with filename <timestamp>-nsxtcombinedhealthnonsos-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 15]
```diff
@@ -2795,6 +2983,12 @@ Function Publish-StorageCapacityHealth {
         .EXAMPLE
         Publish-StorageCapacityHealth -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -rootPass VMw@re1! -workloadDomain sfo-w01
         This example will publish storage usage status for a specific Workload Domain in a VMware Cloud Foundation instance
+
+        .EXAMPLE
+        Publish-StorageCapacityHealth -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -rootPass VMw@re1! -workloadDomain sfo-w01 -outputJson F:\Reports
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2988]:
```suggestion
        Publish-StorageCapacityHealth -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -rootPass VMw@re1! -workloadDomain sfo-w01 -outputJson F:\Reporting
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 16]
```diff
@@ -2795,6 +2983,12 @@ Function Publish-StorageCapacityHealth {
...
+        This example will publish storage usage status for a specific Workload Domain in a VMware Cloud Foundation instance
+        and save it as a json under F:\Reports with filename <timestamp>-storagecapacityhealth-status.json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2990]:
```suggestion
        and saves it as JSON under F:\Reporting with filename <timestamp>-storagecapacityhealth-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 17]
```diff
@@ -4498,6 +4729,88 @@ Function Request-EsxiStorageCapacity {
...
+        This example checks the component connectivity outside of SOS utility for a single Workload Domain in a VMware Cloud Foundation instance.
+
+        .EXAMPLE
+        Publish-ComponentConnectivityHealthNonSOS -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -failureOnly
+        This example checks the component connectivity outside of SOS utility for all Workload Domains across the VMware Cloud Foundation instance but only reports issues.
+
+		.EXAMPLE
+        Publish-ComponentConnectivityHealthNonSOS -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -jsonOutput F:\Reporting
+        This example checks the component connectivity outside of SOS utility for all Workload Domains across the VMware Cloud Foundation instance
+		and saves it as JSON under F:\Reporting\<timestamp>-componentconnectivityhealthnonsos-status.json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:4760]:
```suggestion
		and saves it under F:\Reporting with filename <timestamp>-componentconnectivityhealthnonsos-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 18]
```diff
@@ -8680,6 +8993,33 @@ Function Start-CreateReportDirectory {
 }
 Export-ModuleMember -Function Start-CreateReportDirectory
 
+Function Start-CreateOutputJsonDirectory {
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$jsonFolder,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$jsonFileSuffix
+
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:9000]: Extra line break?

File: VMware.CloudFoundation.Reporting.psm1 [hunk 19]
```diff
@@ -8680,6 +8993,33 @@ Function Start-CreateReportDirectory {
...
+    }
+
+    if (!(Test-Path -Path $jsonFolder)) {
+        New-Item -Path $jsonFolder -ItemType "directory" | Out-Null
+    }
+
+    $jsonDestination
+}
+
+Export-ModuleMember -Function Start-CreateOutputJsonDirectory
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:9021]: Does the need to be exported - looks like it's just an internal function.

File: VMware.CloudFoundation.Reporting.psm1 [hunk 20]
```diff
@@ -2261,14 +2283,20 @@ Function Publish-BackupStatus {
                     }
                 }
 
-                if ($allBackupStatusObject.Count -eq 0) { $addNoIssues = $true }
+                if ($PsBoundParameters.ContainsKey('outputJson')) {
+                    $json = Start-CreateOutputJsonDirectory -jsonFolder $outputJson -jsonFileSuffix $backupJsonSuffix
+                    $allBackupStatusObject | ConvertTo-JSON -depth 10 | Out-File $json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2288]:
```suggestion
                    $allBackupStatusObject | ConvertTo-JSON -Depth 10 | Out-File $json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 21]
```diff
@@ -2335,16 +2370,22 @@ Function Publish-NsxtTransportNodeStatus {
...
                 }
 
-                if ($allNsxtTransportNodeStatusObject.Count -eq 0) {
-                    $addNoIssues = $true
-                }
-                if ($addNoIssues) {
-                    $allNsxtTransportNodeStatusObject = $allNsxtTransportNodeStatusObject | Sort-Object Domain, Resource, Element | ConvertTo-Html -Fragment -PreContent '<a id="nsx-tn"></a><h3>NSX Transport Node Status</h3>' -PostContent '<p>No issues found.</p>'
+                if ($PsBoundParameters.ContainsKey('outputJson')) {
+                    $json = Start-CreateOutputJsonDirectory -jsonFolder $outputJson -jsonFileSuffix $nsxtTransportJsonSuffix
+                    $allNsxtTransportNodeStatusObject | ConvertTo-JSON -depth 10 | Out-File $json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2375]:
```suggestion
                    $allNsxtTransportNodeStatusObject | ConvertTo-JSON -Depth 10 | Out-File $json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 22]
```diff
@@ -2409,19 +2459,24 @@ Function Publish-NsxtTransportNodeTunnelStatus {
...
                     }
                 }
-
-                if ($allNsxtTransportNodeTunnelStatusObject.Count -eq 0) { $addNoIssues = $true }
-                if ($allNsxtTransportNodeTunnelStatusObject.Count -ne 0) {
-                    if ($addNoIssues) {
-                        $allNsxtTransportNodeTunnelStatusObject = $allNsxtTransportNodeTunnelStatusObject | Sort-Object Domain, Resource, Element | ConvertTo-Html -Fragment -PreContent '<a id="nsx-tn-tunnel"></a><h3>NSX Transport Node Tunnel Status</h3>' -PostContent '<p>No issues found.</p>'
+                if ($PsBoundParameters.ContainsKey('outputJson')) {
+                    $json = Start-CreateOutputJsonDirectory -jsonFolder $outputJson -jsonFileSuffix $nsxttntunnelJsonSuffix
+                    $allNsxtTransportNodeTunnelStatusObject | ConvertTo-JSON -depth 10 | Out-File $json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2464]:
```suggestion
                    $allNsxtTransportNodeTunnelStatusObject | ConvertTo-JSON -Depth 10 | Out-File $json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 23]
```diff
@@ -2486,19 +2548,24 @@ Function Publish-NsxtTier0BgpStatus {
...
                     }
                 }
-
-                if ($allNsxtTier0BgpStatusObject.Count -eq 0) { $addNoIssues = $true }
-                if ($nsxtTier0BgpStatus.Count -gt 0) {
-                    if ($addNoIssues) {
-                        $allNsxtTier0BgpStatusObject = $allNsxtTier0BgpStatusObject | Sort-Object 'NSX Manager', 'Domain', 'Tier-0 ID', 'Source Address' | ConvertTo-Html -Fragment -PreContent '<a id="nsx-t0-bgp"></a><h3>NSX Tier-0 Gateway BGP Status</h3>' -PostContent '<p>No issues found.</p>'
+                if ($PsBoundParameters.ContainsKey('outputJson')) {
+                    $json = Start-CreateOutputJsonDirectory -jsonFolder $outputJson -jsonFileSuffix $nsxttier0bgpJsonSuffix
+                    $allNsxtTier0BgpStatusObject | ConvertTo-JSON -depth 10 | Out-File $json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2553]:
```suggestion
                    $allNsxtTier0BgpStatusObject | ConvertTo-JSON -Depth 10 | Out-File $json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 24]
```diff
@@ -2583,14 +2658,21 @@ Function Publish-SnapshotStatus {
                     }
                 }
 
-                if ($allSnapshotStatusObject.Count -eq 0) { $addNoIssues = $true }
-                if ($addNoIssues) {
-                    $allSnapshotStatusObject = $allSnapshotStatusObject | Sort-Object Component, Resource, Element | ConvertTo-Html -Fragment -PreContent '<a id="infra-snapshot"></a><h3>Snapshot Status</h3>' -PostContent '<p>No issues found.</p>'
+                if ($PsBoundParameters.ContainsKey('outputJson')) {
+                    $json = Start-CreateOutputJsonDirectory -jsonFolder $outputJson -jsonFileSuffix $snapshotJsonSuffix 
+                    $allSnapshotStatusObject | ConvertTo-JSON -depth 10 | Out-File $json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2663]:
```suggestion
                    $allSnapshotStatusObject | ConvertTo-JSON -Depth 10 | Out-File $json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 25]
```diff
@@ -2678,22 +2767,118 @@ Function Publish-LocalUserExpiry {
...
         }
-
-        if ($allPasswordExpiryObject.Count -eq 0) { $addNoIssues = $true }
-        if ($addNoIssues) {
-            $allPasswordExpiryObject = $allPasswordExpiryObject | Sort-Object Resource, Component | ConvertTo-Html -Fragment -PreContent '<a id="security-password"></a><h3>Password Expiry Health Status</h3>' -PostContent '<p>No issues found.</p>'
+        if ($PsBoundParameters.ContainsKey('outputJson')) {
+            $json = Start-CreateOutputJsonDirectory -jsonFolder $outputJson -jsonFileSuffix $localuserexpiryJsonSuffix
+			Write-Output $json
+			Write-Output $$allPasswordExpiryObject
+            $allPasswordExpiryObject | ConvertTo-JSON -depth 10 | Out-File $json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2774]:
```suggestion
            $allPasswordExpiryObject | ConvertTo-JSON -Depth 10 | Out-File $json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 26]
```diff
@@ -2678,22 +2767,118 @@ Function Publish-LocalUserExpiry {
...
+            $nsxtVidmStatus = Request-NsxtVidmStatus -server $server -user $user -pass $pass -domain $workloadDomain -failureOnly; $allNsxtHealthObject += $nsxtVidmStatus
+            $nsxtComputeManagerStatus = Request-NsxtComputeManagerStatus -server $server -user $user -pass $pass -domain $workloadDomain -failureOnly; $allNsxtHealthObject += $nsxtComputeManagerStatus
+        } elseif ($PsBoundParameters.ContainsKey("workloadDomain")) {
+            $nsxtVidmStatus = Request-NsxtVidmStatus -server $server -user $user -pass $pass -domain $workloadDomain; $allNsxtHealthObject += $nsxtVidmStatus
+            $nsxtComputeManagerStatus = Request-NsxtComputeManagerStatus -server $server -user $user -pass $pass -domain $workloadDomain; $allNsxtHealthObject += $nsxtComputeManagerStatus
+        }
+
+        if ($PsBoundParameters.ContainsKey("outputJson")) {
+            $json = Start-CreateOutputJsonDirectory -jsonFolder $outputJson -jsonFileSuffix $nsxtCombinedHealthNonSOSJsonSuffix
+            $allNsxtHealthObject | ConvertTo-JSON -depth 10 | Out-File $json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2861]:
```suggestion
            $allNsxtHealthObject | ConvertTo-JSON -Depth 10 | Out-File $json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 27]
```diff
@@ -2804,68 +2998,105 @@ Function Publish-StorageCapacityHealth {
...
-                $sddcManagerStorageHealth = Convert-CssClass -htmldata $sddcManagerStorageHealth; $allStorageCapacityHealth += $sddcManagerStorageHealth
+                if ($PsBoundParameters.ContainsKey('outputJson')) {
+                    $json = Start-CreateOutputJsonDirectory -jsonFolder $outputJson -jsonFileSuffix $storageCapacityHealthJsonSuffix
+                    $combinedjson = @{
+                        "sddc-manager" = $sddcManagerStorageHealth
+                        "vcenter" = $allVcenterStorageHealth
+                        "esxi" = $allEsxiStorageCapacity
+                        "datastore" = $allDatastoreStorageCapacity
+                    }
+                    $combinedJson | ConvertTo-JSON -depth 10 | Out-File $json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:3054]:
```suggestion
                    $combinedJson | ConvertTo-JSON -Depth 10 | Out-File $json
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 28]
```diff
@@ -4498,6 +4729,88 @@ Function Request-EsxiStorageCapacity {
...
+                $NsxtConnectivity = Request-NsxtAuthentication -server $server -user $user -pass $pass -alldomains; $allConnectivityObject += $NsxtConnectivity
+            } else {
+                $vcenterConnectivity = Request-VcenterAuthentication -server $server -user $user -pass $pass -workloadDomain $workloadDomain; $allConnectivityObject += $vcenterConnectivity
+                $NsxtConnectivity = Request-NsxtAuthentication -server $server -user $user -pass $pass -workloadDomain $workloadDomain; $allConnectivityObject += $NsxtConnectivity
+            }
+        }
+
+        if ($PsBoundParameters.ContainsKey("outputJson")) {
+            $json = Start-CreateOutputJsonDirectory -jsonFolder $outputJson -jsonFileSuffix $ComponentConnectivityHealthNonSOSJsonSuffix
+            $allConnectivityObject | ConvertTo-JSON -depth 10 | Out-File $json
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:4795]:
```suggestion
            $allConnectivityObject | ConvertTo-JSON -Depth 10 | Out-File $json
```

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
Minor changes requestion on first pass:

- Other examples and the README.md use `F:\Reporting` vs `F:\Reports`. Added suggestions to keep them consistent.
- Use `-Depth` vs `-depth`.
- Minor grammatical suggestions.

File: VMware.CloudFoundation.Reporting.psm1 (hunk 19 above)
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:9021]:
```suggestion
```

File: VMware.CloudFoundation.Reporting.psm1 (hunk 2 above)
Comment (bhumitra) [reply VMware.CloudFoundation.Reporting.psm1:2236]:
```suggestion
        and saves it under F:\Reporting with filename <timestamp>-backup-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 (hunk 4 above)
Comment (bhumitra) [reply VMware.CloudFoundation.Reporting.psm1:2336]:
```suggestion
        and saves it under F:\Reporting with filename <timestamp>-nsxttransportnode-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 (hunk 2 above)
Comment (bhumitra) [inline VMware.CloudFoundation.Reporting.psm1:2236]:
```suggestion
        and saves it under F:\Reporting with filename <timestamp>-backup-status.json
```

File: VMware.CloudFoundation.Reporting.psm1 (hunk 18 above)
Comment (bhumitra) [inline VMware.CloudFoundation.Reporting.psm1:9000]:
```suggestion
```

File: VMware.CloudFoundation.Reporting.psm1 [hunk 29]
```diff
@@ -2331,9 +2331,9 @@ Function Publish-NsxtTransportNodeStatus {
         This example will publish the BGP status for the NSX transport nodes in a VMware Cloud Foundation instance for a workload domain named sfo-w01.
 
         .EXAMPLE
-        Publish-NsxtTransportNodeStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains
+        Publish-NsxtTransportNodeStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reports
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2334]: Reporting

**Comment by tenthirtyam** [comment]
Thanks for committing the review suggestions. 

I’ll give in a flurry of tests tomorrow and report back.

File: VMware.CloudFoundation.Reporting.psm1 [hunk 30]
```diff
@@ -2301,14 +2329,21 @@ Function Publish-NsxtTransportNodeStatus {
         .EXAMPLE
         Publish-NsxtTransportNodeStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -workloadDomain sfo-w01
         This example will publish the BGP status for the NSX transport nodes in a VMware Cloud Foundation instance for a workload domain named sfo-w01.
+
+        .EXAMPLE
+        Publish-NsxtTransportNodeStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reports
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:2334]:
```suggestion
        Publish-NsxtTransportNodeStatus -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -allDomains -outputJson F:\Reporting
```

**Review by tenthirtyam — APPROVED** [review]
LGTM! 🚀

**Pull request merged**